code/script.js

```diff
@@ -247,97 +300,186 @@ const generateBoard = () => {
+          )
+          charactersInPlay= charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value);
+      } else {
+        alert(
+          `No, the person doesn't have a ${currentQuestion.category}!`
+        )
+        charactersInPlay= charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value);
+      }
+    } else if (currentQuestion.category === 'clothing') {
+      if (keep === true) {
+        alert(
+          `Yes, the person has a ${currentQuestion.value} colored shirt!`
+          )
+          charactersInPlay= charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value);
+      } else {
+        alert(
+          `No, the person doesn't have a ${currentQuestion.value} colored shirt!`
+        )
+        charactersInPlay= charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value);
+      }  
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
-    }
+      if (keep === true) {
+        alert(
+          `Yes, the person has ${currentQuestion.value} ${currentQuestion.category}!`
+        )
+        charactersInPlay= charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value);
+      } else {
+        alert(
+          `No, the person doesn't have ${currentQuestion.value} ${currentQuestion.category}!`
+        )
+        charactersInPlay= charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value);
+      }
   }
-
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
-  // Invoke a function to redraw the board with the remaining people.
+  generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  let confirmation = confirm("Are you sure you want to make a guess?");
+  if (confirmation === true) {
+    checkMyGuess(suspect)
+  }
+};
 
-// If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
-  // 1. Check if the suspect is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (suspect === secret.name) {
+    winOrLoseText.innerHTML = "You won!!! Congrats!"
+    winOrLose.style.display = "flex";
+  } else {
+    winOrLose.style.display = "flex";
+    winOrLoseText.innerHTML = "Game over... Better luck next time."
+  }
+
```

Comment:
In either scenarios the code is applying this: `winOrLose.style.display = "flex";`. It can be moved at the end.
```suggestion
  if (suspect === secret.name) {
    winOrLoseText.innerHTML = "You won!!! Congrats!"
  } else {
    winOrLoseText.innerHTML = "Game over... Better luck next time."
  }
  winOrLose.style.display = "flex";
```